code/datums/components/swarming.dm

```diff
@@ -5,12 +5,22 @@
 	var/list/swarm_members = list()
 
 /datum/component/swarming/Initialize(max_x = 24, max_y = 24)
+	if(!ismovableatom(parent))
+		return COMPONENT_INCOMPATIBLE
 	offset_x = rand(-max_x, max_x)
 	offset_y = rand(-max_y, max_y)
 
 	RegisterSignal(parent, COMSIG_MOVABLE_CROSSED, .proc/join_swarm)
 	RegisterSignal(parent, COMSIG_MOVABLE_UNCROSSED, .proc/leave_swarm)
 
+/datum/component/swarming/Destroy()
+	for(var/datum/component/swarming/other_swarm in swarm_members)
```

Comment:
The loop doesn't need to be typed for this

Reply:
Do you mean it like this?
```
for(var/other_swarm in swarm_members)
```
Because that doesn't work at all.

Reply:
You do it like this:
`
for(var/other in swarm_members)
    var/datum/component/swarming/other_swarm = other
`

Because BYOND basically calls istype() on everything in the list if you typecast the for loop, but that's pointless. Everything in the list is already that type. Frankly it's a bit of a microoptimization except in really large loops (the MC, mostly), but it is good practice.

Reply:
Oh ok. Thanks.